orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/graph/transform/DefaultVariableAssignmentTransform.java

```diff
@@ -59,6 +60,36 @@ public void visitPipelineTemplate(PipelineTemplate pipelineTemplate) {
       throw new IllegalTemplateConfigurationException("Missing variable values for: " + StringUtils.join(missingVariables, ", "));
     }
 
-    // TODO rz - validate variable values match the defined variable type
+    // collect variables where value type doesn't match the required type
+    List<String> wrongTypeErrorMessages = pipelineTemplateVariables.stream()
+      .filter(templateVar -> {
+        if (missingVariables.contains(templateVar.getName())) {
```

Comment:
This condition will never evaluate to true.

Reply:
Removing.